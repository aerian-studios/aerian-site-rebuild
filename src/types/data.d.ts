import * as React from "react";

export const isImageSharp = (
    image: ImageSharp | string
): image is ImageSharp => {
    return typeof image !== "string";
};

export interface Image {
    alt?: string;
    image: string | ImageSharp;
}

export interface ImageSharp {
    childImageSharp: {
        sizes: ImageSharpSizes;
        src: string;
    };
}

export interface ImageSharpSizes {
    aspectRatio?: number;
    sizes?: string;
    src: string;
    srcSet?: string;
    tracedSVG?: string;
}
