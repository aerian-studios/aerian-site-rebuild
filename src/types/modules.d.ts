declare module "*.svg";
declare module "*.png";
declare module "*.json";

// For storyshots
declare module "babel-plugin-require-context-hook/register";

// for maps
declare module "pigeon-overlay";
declare module "pigeon-maps";
