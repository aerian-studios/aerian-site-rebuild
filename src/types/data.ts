import { Stats } from "fs";
import { ParsedPath } from "path";

export type ImageField = string | ImageSharp;
export type FileField = string | Partial<FileNode>;

// Images

export interface GalleryImage {
    alt?: string;
    image: ImageField;
}

export interface FileNode extends ParsedPath, Stats {
    publicURL: string;
    id: string;
    children: any[];
    sourceInstanceName: string;
    absolutePath: string;
    relativePath: string;
    relativeDirectory: string;
    extension: string;
    size: number;
    prettySize: string;

    modifiedTime: string;
    accessTime: string;
    changeTime: string;
    birthTime: string;
    internal: {
        contentDigest: string;
        type: "Directory" | "File";
        mediaType?: string;
        description: string;
    };
}

export interface ImageSharp extends Partial<FileNode> {
    childImageSharp: {
        fluid?: ImageSharpFluid;
        fixed?: ImageSharpFixed;
    };
}
export interface ImageSharpFluid {
    aspectRatio?: number;
    sizes?: string;
    src: string;
    srcSet?: string;
    tracedSVG?: string;
    base64?: string;
}

export interface ImageSharpFixed {
    aspectRatio?: number;
    src: string;
    srcSet?: string;
    tracedSVG?: string;
    base64?: string;
    width: number;
    height: number;
}

// PAGES

export interface NodeList<T> {
    edges: Array<Edge<T>>;
}

export interface Edge<T> {
    node: T;
}
export interface PagesListData {
    data: {
        allPagesJson: PageList;
    };
}

export interface PageList {
    edges: PageListNode[];
}

export type PageListNode = Edge<PageRef>;

export interface PageRef {
    id: string;
    path: string;
    title: string;
}

export interface PageSection {
    title: string;
    image?: ImageField;
    smallImage?: ImageField;
    subtitle: string;
    blurb: string;
    activities?: Activity[];
    successStory?: SuccessStory;
    testimonial?: Testimonial;
}

export interface Page {
    title: string;
    path: string;
    heroImage?: ImageField;
    subheading: string;
    description: string;
    seoTitle: string;
    seoDescription: string;
    seoKeywords: string;
}

export interface WhatWeDo extends Page {
    sections?: PageSection[];
}

export interface MeetTheTeam extends Page {
    staff: Staff[];
}

export interface About extends Page {
    infographic?: Infographic[];
}

export type InfographicType =
    | "Single Line"
    | "Compact Single Line"
    | "Split"
    | "Bar"
    | "Vertical"
    | "Horizontal"
    | "Surround";
export interface Infographic {
    primaryText?: string;
    type: InfographicType;
    secondaryText?: string;
    image?: ImageField;
    imageCount: number;
    secondaryImage?: ImageField;
}

export interface Staff {
    name: string;
    jobTitle: string;
    live: boolean;
    imageNormal: ImageField;
    imageFunny: ImageField;
    description: string;
    fact: string;
    skills: string[];
}

export interface ReactRouterLocation {
    action: string;
    hash: string;
    key: string;
    pathname: string;
    search: string;
    state: null | string;
}

export interface Testimonial {
    quote?: string;
    person?: string;
    title?: string;
    avatar?: ImageField;
    image?: ImageField;
}

export interface Stat {
    title: string;
    text: string;
    image: ImageField;
}
export interface SuccessStory {
    title: string;
    stats?: Stat[];
}

export interface Activity {
    title: string;
    text: string;
}

export interface Hurdle {
    title: string;
    image: ImageField;
    text: string;
}

export interface ProjectStage {
    description: string;
    hurdles: Hurdle[];
}

export interface Performance {
    title: string;
    text: string;
}
export interface ProjectBox {
    name: string;
    slug: string;
    heroImage: ImageField;
    thumbnail: ImageField;
    client: Client;
    titleLineOne: string;
}
export interface Project {
    name: string;
    slug: string;
    heroImage: ImageField;
    heroVideo?: string;
    titleLineOne: string;
    titleLineTwo?: string;
    client: Client;
    goLiveDate?: string;
    caseStudyTitle: string;
    caseStudyText: string;
    caseStudyImage: ImageField;
    externalUrl: string;
    rolloverDetails: string;
    homepage: boolean;
    featured: boolean;
    gallery: GalleryImage[];
    challenge: ProjectStage;
    solution: ProjectStage;
    results: ProjectStage;
    performance?: Performance[];
    testimonial?: Testimonial;
}

export interface Client {
    name: string;
    logo: ImageField;
    promoLogo: ImageField;
    featured: boolean;
    featuredProject?: string;
    id: string;
}
