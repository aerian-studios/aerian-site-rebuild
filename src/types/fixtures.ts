import {
    About,
    Client,
    ImageField,
    ImageSharp,
    Infographic,
    MeetTheTeam,
    PageSection,
    Performance,
    Project,
    ProjectBox,
    SuccessStory,
    Testimonial,
    WhatWeDo
} from "./data";

import * as aboutData from "../pages/about.json";
import * as clientData from "../pages/clients/cnn.json";
import * as meetTheTeamData from "../pages/meet-the-team.json";
import * as projectData from "../pages/projects/bbc-iwonder-perfect-diet-for-you.json";
import * as whatWeDoData from "../pages/what-we-do.json";

import { absolutifyURLs } from "../lib/helpers";
export const client = absolutifyURLs<Client>(clientData);

export const project = { ...absolutifyURLs<Project>(projectData), client };
export const projectBox: ProjectBox = {
    name: project.name,
    slug: project.slug,
    titleLineOne: project.titleLineOne,
    thumbnail: project.heroImage,
    heroImage: project.heroImage,
    client
};
export const whatWeDo = absolutifyURLs<Required<WhatWeDo>>(whatWeDoData);
export const about = absolutifyURLs<About>(aboutData);
export const meetTheTeam = absolutifyURLs<MeetTheTeam>(meetTheTeamData);
export const projectStage = project.challenge;
export const hurdles = projectStage.hurdles;
export const hurdle = hurdles[0];
export const performances: Performance[] = project.performance as Performance[];
export const performance = performances[0];
export const testimonial: Testimonial = project.testimonial as Testimonial;
export const testimonialSection: PageSection = whatWeDo.sections[0];
export const successStorySection: PageSection = whatWeDo.sections[1];
export const successStory: SuccessStory = successStorySection.successStory as SuccessStory;
export const pageSection = {
    ...successStorySection,
    testimonial: testimonialSection.testimonial
} as Required<PageSection>;
export const activity = pageSection.activities[0];
export const stat = successStory.stats![0];
export const staff = meetTheTeam.staff;
export const person = staff[0];
export const infographics = about.infographic as Infographic[];
export const infographic = infographics[0];

export const image: string = person.imageFunny as string;
export const imageSharpFluid: ImageSharp = {
    childImageSharp: {
        fluid: {
            aspectRatio: 3 / 4,
            sizes: "(max-width: 290px) 100vw, 290px",
            base64:
                "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAYCAYAAAD6S912AAAACXBIWXMAAAsSAAALEgHS3X78AAAEe0lEQVQ4y42Te1BUVRzH70z/9V9ZToXy2GXfD3Z57C4sDyHtreWMIqmEgUVCak4qwiiRiausiIhk6JLiAzXGEE1FBVMaEBtsrJnQwsEaNTMbmxRrFJZP516EWXykZ+Y3d/fc7/mc7+9xJQJWf3+/8vT7/dy9BvdkzaDufkt6EKznxnXKS0uo+rSCP69eVfb6+vqGaf8XGCjs6enhtZdeRJIkHn9MwmGzcv589yNBhzns7e1VfnuKP2GcTY8nO51Vs96iaNp4ctImcvn3K0NZPBJQXn9fv8EHGWkcKfuIrrpqftmzlXO1VVTNnUG1SD/Q5QOBMszvHxB9297G4sxUjlV5+bqylK76Lzi2bjUNxflsWeOhr1858HCH/ju3fnP8KMU502mqWMmmRR/Sss7L/pKl1OTPY/daDzf/+ffhNRxwONDd002H2LDofVrWe2kuW8FRz8ecKPPQXuGhtVwAL10cAN5ntIYBBwW/7d9Hw7ICftjpo2tLJY1FBTQVL6FzUyXt4oKe7u6HAwMFt8SBI6XLOSAcndpcSUt5CQ3zFnCgpJDmSi83/7r2aCkHujwrGrPLs4RD65bTurGcVpHyHs9CTjTsHLr8QV/M8MG+A7z2xxWqlxbgK5zHcd9qmtcuY33eLH463aG87+29PQS7+ykF3jRws+j47Vs0VJRQNieLls1llOZm8NnC9zjbcXKYG3ke5Qgc9CHg4At5nTzZQXVRPnVF8+ms89HiK6XN52V1QR4rSsro7Dxzz3APQiX5RWAtjhxuIjnlZTInTaamYDY/1lXx84FaOr+spjAnmxFPP4fZFMHENyZTXLySxoOHuXDh4r01bGtrJzPzXWKi43A43FgskazJn0tn/Qa6m3fzff1WslJT0WgNGAwmgoPDGD0qjHC1AacjnqKiZcK1H2n79h1kZ+cSGelArzPhcsUTF5vAE089S17GJM41bKTr8C6+27uT1PGvowrXodHIoUevN2G1RmI22xUjly5dRgoOVhEaqhab1jsRgcVqEw5C8eXN5PzeDZxp3MHBmhqy3pxKiNBqwrWo1Vq0wq3ZbBPQKGEiidraXUgGgxk5jEazkoocJgFOco/h88X5/LpvG6f2bmP/1t3MnpZOmFqDTmdU3JlMVgVoMgmoJUqUKh5pVFCIIpChOhFarR6bLYqEhOdxu8eSnTaDqhVeWr86Rm5WNuEiXaPJImpsE7oY7CJkt0aDlVhXItKIJ0diMloYHaLmnSmT8M6fI+oRS0J8ihIOIYpzp7AwdwFZU6djEFr5Qrs9huiYWHE2QpTAIHrgxBHjRhopxkCl1vNCUhKN61fx9pQ0YoT1ROEw3p3MmKRxxMYmKk6S3fHY7NFKA+WQG6JSaZS0o6JcRAsj0jMjgwgarSLt1VdYPDODUDEGcoFdrgSRdgqJiWPR6S2kTUwlfcIELBGRwkmccOhQQGGh4ei0RsVhpN2JFBoSTliYFrslgmSnUznsFHPociYoMLkuKpWWKJGiS4BstmglXbmzBoOFoKBgAdUQIf7L8R+bum8HTMuX7wAAAABJRU5ErkJggg==",
            src: image,
            srcSet: image
        }
    }
};

export const imageSharpFixed: ImageSharp = {
    childImageSharp: {
        fixed: {
            width: 290,
            height: 290,
            base64:
                "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAYCAYAAAD6S912AAAACXBIWXMAAAsSAAALEgHS3X78AAAEe0lEQVQ4y42Te1BUVRzH70z/9V9ZToXy2GXfD3Z57C4sDyHtreWMIqmEgUVCak4qwiiRiausiIhk6JLiAzXGEE1FBVMaEBtsrJnQwsEaNTMbmxRrFJZP516EWXykZ+Y3d/fc7/mc7+9xJQJWf3+/8vT7/dy9BvdkzaDufkt6EKznxnXKS0uo+rSCP69eVfb6+vqGaf8XGCjs6enhtZdeRJIkHn9MwmGzcv589yNBhzns7e1VfnuKP2GcTY8nO51Vs96iaNp4ctImcvn3K0NZPBJQXn9fv8EHGWkcKfuIrrpqftmzlXO1VVTNnUG1SD/Q5QOBMszvHxB9297G4sxUjlV5+bqylK76Lzi2bjUNxflsWeOhr1858HCH/ju3fnP8KMU502mqWMmmRR/Sss7L/pKl1OTPY/daDzf/+ffhNRxwONDd002H2LDofVrWe2kuW8FRz8ecKPPQXuGhtVwAL10cAN5ntIYBBwW/7d9Hw7ICftjpo2tLJY1FBTQVL6FzUyXt4oKe7u6HAwMFt8SBI6XLOSAcndpcSUt5CQ3zFnCgpJDmSi83/7r2aCkHujwrGrPLs4RD65bTurGcVpHyHs9CTjTsHLr8QV/M8MG+A7z2xxWqlxbgK5zHcd9qmtcuY33eLH463aG87+29PQS7+ykF3jRws+j47Vs0VJRQNieLls1llOZm8NnC9zjbcXKYG3ke5Qgc9CHg4At5nTzZQXVRPnVF8+ms89HiK6XN52V1QR4rSsro7Dxzz3APQiX5RWAtjhxuIjnlZTInTaamYDY/1lXx84FaOr+spjAnmxFPP4fZFMHENyZTXLySxoOHuXDh4r01bGtrJzPzXWKi43A43FgskazJn0tn/Qa6m3fzff1WslJT0WgNGAwmgoPDGD0qjHC1AacjnqKiZcK1H2n79h1kZ+cSGelArzPhcsUTF5vAE089S17GJM41bKTr8C6+27uT1PGvowrXodHIoUevN2G1RmI22xUjly5dRgoOVhEaqhab1jsRgcVqEw5C8eXN5PzeDZxp3MHBmhqy3pxKiNBqwrWo1Vq0wq3ZbBPQKGEiidraXUgGgxk5jEazkoocJgFOco/h88X5/LpvG6f2bmP/1t3MnpZOmFqDTmdU3JlMVgVoMgmoJUqUKh5pVFCIIpChOhFarR6bLYqEhOdxu8eSnTaDqhVeWr86Rm5WNuEiXaPJImpsE7oY7CJkt0aDlVhXItKIJ0diMloYHaLmnSmT8M6fI+oRS0J8ihIOIYpzp7AwdwFZU6djEFr5Qrs9huiYWHE2QpTAIHrgxBHjRhopxkCl1vNCUhKN61fx9pQ0YoT1ROEw3p3MmKRxxMYmKk6S3fHY7NFKA+WQG6JSaZS0o6JcRAsj0jMjgwgarSLt1VdYPDODUDEGcoFdrgSRdgqJiWPR6S2kTUwlfcIELBGRwkmccOhQQGGh4ei0RsVhpN2JFBoSTliYFrslgmSnUznsFHPociYoMLkuKpWWKJGiS4BstmglXbmzBoOFoKBgAdUQIf7L8R+bum8HTMuX7wAAAABJRU5ErkJggg==",
            src: image,
            srcSet: image
        }
    }
};
export const imageSharp = imageSharpFluid;
export const imageField: ImageField = imageSharp;
