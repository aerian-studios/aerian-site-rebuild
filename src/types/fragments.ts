import { graphql } from "gatsby";

export const pageFieldsFragment = graphql`
    fragment PageFields on PagesJson {
        title
        heroImage {
            childImageSharp {
                fluid(srcSetBreakpoints: [1024, 720, 360]) {
                    ...GatsbyImageSharpFluid_withWebp
                }
            }
        }
        subheading
        description
        seoTitle
        seoDescription
        seoKeywords
    }
`;

export const clientFragment = graphql`
    fragment Client on ClientsJson {
        name
        logo {
            childImageSharp {
                fluid(maxWidth: 600, srcSetBreakpoints: [420, 210]) {
                    ...GatsbyImageSharpFluid_withWebp_tracedSVG
                }
            }
            publicURL
        }
        promoLogo {
            childImageSharp {
                fluid(maxWidth: 600, srcSetBreakpoints: [420, 210]) {
                    ...GatsbyImageSharpFluid_withWebp_tracedSVG
                }
            }
            publicURL
        }
        featured
        featuredProject
    }
`;

export const projectBoxFragment = graphql`
    fragment ProjectBox on ProjectsJson {
        name
        slug
        titleLineOne
        client {
            ...Client
        }
        heroImage {
            childImageSharp {
                fluid(
                    maxWidth: 1066
                    maxHeight: 1066
                    srcSetBreakpoints: [1066, 720, 360, 250]
                ) {
                    ...GatsbyImageSharpFluid_withWebp_noBase64
                }
            }
            publicURL
        }
    }
`;

export const projectFragment = graphql`
    fragment Project on ProjectsJson {
        name
        slug
        client {
            ...Client
        }
        titleLineOne
        titleLineTwo
        goLiveDate
        caseStudyTitle
        caseStudyText
        externalUrl
        heroImage {
            childImageSharp {
                fluid(srcSetBreakpoints: [1024, 720, 360]) {
                    ...GatsbyImageSharpFluid_withWebp
                }
            }
            publicURL
        }
        heroVideo {
            publicURL
        }
        gallery {
            image {
                childImageSharp {
                    fluid(srcSetBreakpoints: [720, 360]) {
                        ...GatsbyImageSharpFluid_withWebp
                    }
                }
                publicURL
            }
            alt
        }
        challenge {
            description
            hurdles {
                title
                image {
                    childImageSharp {
                        fixed(width: 290) {
                            ...GatsbyImageSharpFixed_withWebp_tracedSVG
                        }
                    }
                    publicURL
                }
                text
            }
        }
        solution {
            hurdles {
                title
                image {
                    childImageSharp {
                        fixed(width: 290) {
                            ...GatsbyImageSharpFixed_withWebp_tracedSVG
                        }
                    }
                    publicURL
                }
                text
            }
            description
        }
        results {
            description
            hurdles {
                title
                image {
                    childImageSharp {
                        fixed(width: 290) {
                            ...GatsbyImageSharpFixed_withWebp_tracedSVG
                        }
                    }
                    publicURL
                }
                text
            }
        }
        performance {
            title
            text
        }
        testimonial {
            quote
            person
            title
            avatar {
                childImageSharp {
                    fixed(width: 75) {
                        ...GatsbyImageSharpFixed_withWebp_tracedSVG
                    }
                }
                publicURL
            }
        }
        id
    }
`;
