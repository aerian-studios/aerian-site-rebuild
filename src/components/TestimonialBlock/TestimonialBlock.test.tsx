/// <reference types="@types/jest" />
import * as React from "react";

import * as renderer from "react-test-renderer";

import { testimonial } from "../../types/fixtures";
import { TestimonialBlock } from "./index";
const noImageTestimonial = { ...testimonial, image: undefined }

describe("TestimonialBlock", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<TestimonialBlock className="myClass" testimonial={testimonial} />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("renders correctly without an image", () => {
        const tree = renderer
            .create(<TestimonialBlock className="myClass" testimonial={noImageTestimonial} />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
});