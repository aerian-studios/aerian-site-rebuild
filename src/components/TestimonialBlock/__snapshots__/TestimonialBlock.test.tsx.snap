// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestimonialBlock renders correctly 1`] = `
<div
  className=" myClass"
  itemProp="review"
  itemScope={true}
  itemType="http://schema.org/Review"
>
  <div
    className=""
  >
    <img
      src="https://raw.githubusercontent.com/aerian-studios/aerian-site-rebuild/master/static/assets/imported/281/bbc-perfect-diet-for-you-qoute-image.jpg.jpg"
    />
  </div>
  <blockquote>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-quote-left fa-w-16 "
      data-icon="quote-left"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 256h-80v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8c-88.4 0-160 71.6-160 160v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zm-288 0H96v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8C71.6 32 0 103.6 0 192v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      itemProp="reviewBody"
    >
      It was seen by 1.5m unique browsers and was really well received by audience members. Users wrote in to notify us of their success losing weight having taken the test, and then the onward journey to BBC Food recipes. We couldnâ€™t have done it without Aerian.
    </div>
    <footer>
      <figure>
        <div
          className=""
        >
          <img
            src="https://raw.githubusercontent.com/aerian-studios/aerian-site-rebuild/master/static/assets/imported/282/anna.png"
          />
        </div>
      </figure>
      <div>
        <span
          itemProp="author"
        >
           Anna-Louise Taylor
        </span>
        <span
          itemProp="name"
        >
          Senior Content Producer
        </span>
      </div>
    </footer>
  </blockquote>
</div>
`;
